pkg/model/image_target.go

```diff
@@ -3,19 +3,45 @@ package model
 import (
 	"fmt"
 
+	"github.com/docker/distribution/reference"
+	"github.com/google/go-cmp/cmp"
+
 	"github.com/tilt-dev/tilt/internal/container"
 	"github.com/tilt-dev/tilt/internal/sliceutils"
 	"github.com/tilt-dev/tilt/pkg/apis"
 	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
 )
 
+type LiveUpdateSpec struct {
+	// A KubernetesDiscovery maps deployed objects to pods.
+	//
+	// The live-updater watches the named KubernetesDiscovery object to find pods to
+	// live update into.
+	KubernetesDiscoveryName string
+
+	// ImageSelector specifies the name of the image that we're copying files into.
+	//
+	// The live-updater uses this to figure out which containers to live update.
+	ImageSelector string
+}
+
 type ImageTarget struct {
 	// An apiserver-driven data model for injecting the image into other resources.
 	v1alpha1.ImageMapSpec
+	LiveUpdateSpec
 
 	Refs         container.RefSet
 	BuildDetails BuildDetails
 
+	// In a live-update-only image, we don't inject the image into the Kubernetes
+	// deploy, we only live-update to the deployed object. See this issue:
+	//
+	// https://github.com/tilt-dev/tilt/issues/4577
+	//
+	// This is a hacky way to model this right now until we
+	// firm up how images work in the apiserver.
+	IsLiveUpdateOnly bool
```

Comment:
Doesn't matter much since this is a hack, but is this already represented by `BuildDetails == nil`? (which maybe we should rename to `BuildSpec`) That seems like it'd make the "some targets have live update but not image build and vice versa" thing represented a bit more symmetrically.

Reply:
nope, it can't be nil, because the BuildDetails contains the Live Update steps.

But yeah, this isn't how I would model it from the get-go, but this data model is going to get thrown away anyway so it doesn't really matter.